streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java

```diff
@@ -175,6 +177,12 @@ public void testAddProcessorWithNullParents() {
         builder.addProcessor("processor", new MockProcessorSupplier<>(), (String) null);
     }
 
+    @Test(expected = TopologyException.class)
+    public void testAddProcessorWithBadSupplier() {
```

Comment:
We should also test `addGlobalStateStore()`

Reply:
Assuming you mean `.addGlobalStore()`, as there's no `.addGlobalStateStore()` in the builder. There was a supplier usage check missing there too.